### PR TITLE
Add excludepkgs option to repo command

### PR DESCRIPTION
**Pull request by tmarcu**
Signed-off-by: Tudor Marcu <tudor.marcu@intel.com>

### CODE REVIEW

**Comment by gtkramer** [comment]
@tmarcu  and I pair-programmed this.

+1

File: mixer/cmd/repos.go [hunk 1]
```diff
@@ -66,12 +67,21 @@ var setURLRepoCmd = &cobra.Command{
 	Run:   runSetURLRepo,
 }
 
+var setExcludesRepoCmd = &cobra.Command{
+	Use:   "exclude <repo> <pkg> [<pkg>...]",
+	Short: "Exclude packages from a specified repo",
+	Long:  `Exclude packages from a specified repo. These packages will be ignored during build bundles. Globbing is supported.`,
```
Comment (matthewrsj) [inline mixer/cmd/repos.go]: If you really are adding this functionality to mixer as opposed to letting users modify the yum conf themselves you should probably put the use case in the `Long` text here.

File: builder/builder.go [hunk 2]
```diff
@@ -1840,6 +1840,33 @@ func (b *Builder) SetURLRepo(name, url string) error {
 	return DNFConf.SaveTo(b.Config.Builder.DNFConf)
 }
 
+// SetExcludesRepo sets the ecludes for the repo <name> to [pkgs...]
+func (b *Builder) SetExcludesRepo(name, pkgs string) error {
```
Comment (matthewrsj) [inline builder/builder.go]: Maybe rename `name` to `repoName` to be more explicit about what it is naming. I had to look at the caller to figure out what that was.

**Comment by gtkramer** [comment]
I think this is a feature that would help users guarantee correctness of bundles built by mixer in lieu of some shortcomings with dnf where dnf produces an unexpected dependency solution.  This mainly happens when a user is intending to fully substitute a package from Clear's repo with one from their own.  If anything is wrong with the substitution, a feature like this would expose the issue early on.

Although, I see how this starts to turn mixer into a little bit of a fancy interface for dnf confs.  For non-human users working in a bash environment, editing config files tends to be a pain and somewhat brittle.  Having mixer offer commands to interface with dnf confs like `mixer repo add` are definitely nice and welcomed features for automation.

**Comment by tmarcu** [comment]
Mixer has historically changed the configs it uses and their formats, I think the most user friendly solution is to have them edited by mixer itself, so when they inevitably change again, it's completely supported and transparent to the user since they will still be calling the same "add/edit" commands.

**Comment by matthewrsj** [comment]
> Mixer has historically changed the configs it uses and their formats

It changes the in-house configs. This is a DNF config and we as mixer developers have no control over DNF config format.

> they will still be calling the same "add/edit" commands

Well I see your point, but those were only added because we needed top-level support for remote repos.

**Comment by tmarcu** [comment]
That's true, but it's still less work IMO to just use mixer to say exclude <list> than edit (hidden) confs in scripts and elsewhere.

**Comment by matthewrsj** [comment]
**maybe** the conf shouldn't be hidden, but ok...

File: mixer/cmd/repos.go (hunk 1 above)
Comment (matthewrsj) [reply mixer/cmd/repos.go]: Please still add the use-case here. Something like "Allows the user to explicitly select which packages to use when building bundles by excluding the unwanted ones"

**Comment by tmarcu** [comment]
It's only hidden when generated by default if the user doesn't specify their own conf, it can be named and placed however the user wants.

**Comment by matthewrsj** [comment]
:disappointed:

File: mixer/cmd/repos.go (hunk 1 above)
Comment (matthewrsj) [reply mixer/cmd/repos.go]: you didn't add the use case

Comment (matthewrsj) [inline mixer/cmd/repos.go]: You didn't add the use case

**Pull request merged**